instrumentation-genai/opentelemetry-instrumentation-openai-v2/README.rst

```diff
@@ -17,6 +17,53 @@ Installation
 
     pip install opentelemetry-instrumentation-openai-v2
 
+Usage
+-----
+
+Instrumenting all clients
+*************************
+
+When using the instrumentor, all clients will automatically trace OpenAI chat completion operations
+and capture prompts and completions as events.
+
+Make sure to configure OpenTelemetry tracing, logging, and events to capture all telemetry emitted by the instrumentation.
+
+.. code-block:: python
+
+    from opentelemetry.instrumentation.openai_v2 import OpenAIInstrumentor
+
+    OpenAIInstrumentor().instrument()
+
+    client = OpenAI()
+    response = client.chat.completions.create(
+        model="gpt-4o-mini",
+        messages=[
+            {"role": "user", "content": "Write a short poem on open telemetry."},
+        ],
+    )
+
+Enabling message content
+*************************
+
+Message content such as the contents of the prompt, completion, function arguments and return values
+are not captured by default. To enable message content, set the environment variable
```

Comment:
```suggestion
are not captured by default. To capture message content as log events, set the environment variable
```

Reply:
Is there a doco or something related to setting up log events instrumentation that can be linked in place here? Most existing apps will not have log events provider setup.

Reply:
@karthikscale3 just confirming we are using log events and not span events for this even in the interim?

Reply:
We are using span events in the interim at the moment - https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2759/files?diff=unified&w=0#diff-954647898e3f111e390e80f228b8d86a8df076df8f43496783d6f8660a5f29aaR118

Reply:
@dasiths https://github.com/open-telemetry/opentelemetry-python/issues/4269 has the patch for log event enabling, but that will only work if your backend/collector accepts logs.

Reply:
I'm a bit confused because I'm trying to test it with `OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT=true` and I can see the span but not span events in Tempo.

![image](https://github.com/user-attachments/assets/79d4404f-7ee8-4cb5-ae92-77a098f07137)

Span events for other spans are showing up so it's not an integration issue with Tempo. I fear I may be doing something wrong which might be an opportunity to improve the docs once we figure it out.

Reply:
My bad sorry, i just realized this PR was merged - https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2925/files . Yes log events is what we are using now.

Reply:
@dasiths can you confirm you can see the log events in tempo by clicking "logs for this span"? I've not tried tempo..

Reply:
> @dasiths can you confirm you can see the log events in tempo by clicking "logs for this span"? I've not tried tempo..

Yes the logs do appear once `OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT=true` is set. https://github.com/jsburckhardt/ai-translation-bot-demo/blob/c3ddf39f65cba2eac742fdfbfb4dec4ebe19d7b3/app/services/observability.py#L150C6-L163C13

Reply:
"I did however get a warning saying event logger provider is already set." me, too, but hopefully when "zero code" works that message goes away with it.